### ISSUE TITLE
Wrong indentation error when having a sequence with anchor only

### ISSUE THREAD

**Issue by roymiloh** [adrienverge/yamllint#526]
Hey, the following is failing due to a wrong indentation:

```yaml
root:
  - &list
      - foo: 1
      - bar: 1
```

The error is:

>  3:7       error    wrong indentation: expected 4 but found 6  (indentation)

However, when I turn it into an ordinary sequence, it passes:

```yaml
root:
  - list:
      - foo: 1
      - bar: 1
```

It seems like it starts counting from the hyphen when there's an anchor **only**.

Here's another example that passes:

```yaml
root:
  - foo: &list
      - foo: 1
      - bar: 1
```

The `yamllint` version is `1.28.0`.

**Comment by adrienverge** [comment]
Hello,

This seems normal behavior to me.

I don't think this is related to anchors, you can remove `&list` from your example and you would get the same results. The difference between your failing / passing examples are the use of `list:`. When present, the `foo`-`bar` list is nested inside a mapping object. When absent, this list is not nested, and a direct child of the parent list. In this case, it should be indented at +2 from its parent, indenpendently of the presence of an anchor.

Said differently, your first example:
```yaml
root:
  - &list
      - foo: 1
      - bar: 1
```
is equivalent to:
```yaml
root:
  -   - foo: 1
      - bar: 1
```
which should be indented like this:
```yaml
root:
  - - foo: 1
    - bar: 1
```
or like this:
```yaml
root:
  - &list
    - foo: 1
    - bar: 1
```

Please let me know if this is clear!
